espresso/submitter/submitter.go

```diff
@@ -0,0 +1,218 @@
+package submitter
+
+import (
+	"fmt"
+	"math/big"
+	"time"
+
+	espresso_client "github.com/EspressoSystems/espresso-network/sdks/go/client"
+	espresso_light_client "github.com/EspressoSystems/espresso-network/sdks/go/light-client"
+
+	"github.com/ethereum/go-ethereum/ethdb"
+
+	"github.com/offchainlabs/nitro/arbos/arbostypes"
+	"github.com/offchainlabs/nitro/arbutil"
+	espresso_key_manager "github.com/offchainlabs/nitro/espresso/key-manager"
+	"github.com/offchainlabs/nitro/util/stopwaiter"
+)
+
+type MessageGetter interface {
+	GetMessage(seqNum arbutil.MessageIndex) (*arbostypes.MessageWithMetadata, error)
+}
+
+type EspressoSubmitter interface {
+	Start(sw *stopwaiter.StopWaiter) error
+	NotifyNewPendingMessages(pos arbutil.MessageIndex, messages []arbostypes.MessageWithMetadataAndBlockInfo) error
+	GetKeyManager() espresso_key_manager.EspressoKeyManagerInterface
+
+	IsEscapeHatchEnabled() bool
+	GetLastConfirmedPosition() (*arbutil.MessageIndex, error)
+}
+
+type EspressoSubmitterConfig struct {
+	// Simple Configuration values. These will be expected to have default
+	// values set for them, but can be overridden by the user.
+
+	ChainID                               uint64
+	EspressoTxnsPollingInterval           time.Duration
```

Comment:
do all these values need to be public? we are already creating setter functions for these variables

Reply:
So this is a field on a `Config` struct for creating an `EspressoSubmitter`.  In general this is applying the [Functional Options Pattern](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis) for flexibility and readability.

These fields are currently public to allow for the creation of custom `Options` that are external to the package, should they be needed.  However, ultimately, you are correct with the current spread of functionality, these fields do not need to be explicitly `public`, and could be `private` instead.

However, since the `Config` struct is separate from the actual implementation, these fields being public or private doesn't really matter in my view.

What are your thoughts?

Reply:
Generally speaking its better if we have less public fields in code but its fine if we keep them as public as well.  I dont have strong opinions on this

Reply:
You are correct, this is true generally speaking. However it is important to consider the context under `public` versus `private` and why it's a practice.  The general idea of of having fields being `public` versus `private` has to due with mutability.  , and the  the potential  publicly accessible fields to be modified by external sources / libraries without direct visibility / observability in business logic.

In this specific case, this is a configuration structure that an External user is meant to be able to configure and modify as needed before it gets used / consumed by the actual implementation, at which point it becomes immutable, and can no longer affect the implementation directly / indirectly.